Mathlib/Topology/UniformSpace/AbstractCompletion.lean

```diff
@@ -0,0 +1,405 @@
+  funext fun a => pkg.inductionOn a (isClosed_eq hf hg) h
+#align abstract_completion.funext AbstractCompletion.funext
+
+variable [UniformSpace β]
+
+section Extend
+
+/-- Extension of maps to completions -/
+protected def extend (f : α → β) : hatα → β :=
+  if UniformContinuous f then pkg.denseInducing.extend f else fun x => f (pkg.dense.some x)
+#align abstract_completion.extend AbstractCompletion.extend
+
+variable {f : α → β}
+
+theorem extend_def (hf : UniformContinuous f) : pkg.extend f = pkg.denseInducing.extend f :=
+  if_pos hf
+#align abstract_completion.extend_def AbstractCompletion.extend_def
+
+theorem extend_coe [T2Space β] (hf : UniformContinuous f) (a : α) : (pkg.extend f) (ι a) = f a := by
+  rw [pkg.extend_def hf]
+  exact pkg.denseInducing.extend_eq hf.continuous a
+#align abstract_completion.extend_coe AbstractCompletion.extend_coe
+
+variable [CompleteSpace β]
+
+theorem uniformContinuous_extend : UniformContinuous (pkg.extend f) := by
+  by_cases hf : UniformContinuous f
+  · rw [pkg.extend_def hf]
+    exact uniformContinuous_uniformly_extend pkg.uniformInducing pkg.dense hf
+  · change UniformContinuous (ite _ _ _)
+    rw [if_neg hf]
+    exact uniformContinuous_of_const fun a b => by congr 1
+#align abstract_completion.uniform_continuous_extend AbstractCompletion.uniformContinuous_extend
+
+theorem continuous_extend : Continuous (pkg.extend f) :=
+  pkg.uniformContinuous_extend.continuous
+#align abstract_completion.continuous_extend AbstractCompletion.continuous_extend
+
+variable [SeparatedSpace β]
+
+theorem extend_unique (hf : UniformContinuous f) {g : hatα → β} (hg : UniformContinuous g)
+    (h : ∀ a : α, f a = g (ι a)) : pkg.extend f = g := by
+  apply pkg.funext pkg.continuous_extend hg.continuous
+  simpa only [pkg.extend_coe hf] using h
+#align abstract_completion.extend_unique AbstractCompletion.extend_unique
+
+@[simp]
+theorem extend_comp_coe {f : hatα → β} (hf : UniformContinuous f) : pkg.extend (f ∘ ι) = f :=
+  funext fun x =>
+    pkg.inductionOn x (isClosed_eq pkg.continuous_extend hf.continuous) fun y =>
+      pkg.extend_coe (hf.comp <| pkg.uniformContinuous_coe) y
+#align abstract_completion.extend_comp_coe AbstractCompletion.extend_comp_coe
+
+end Extend
+
+section MapSec
+
+variable (pkg' : AbstractCompletion β)
+
+-- mathport name: exprhatβ
+local notation "hatβ" => pkg'.space
+
+-- mathport name: exprι'
+local notation "ι'" => pkg'.coe
+
+/-- Lifting maps to completions -/
+protected def map (f : α → β) : hatα → hatβ :=
+  pkg.extend (ι' ∘ f)
+#align abstract_completion.map AbstractCompletion.map
+
+-- mathport name: exprmap
+local notation "map" => pkg.map pkg'
+
+variable (f : α → β)
+
+theorem uniformContinuous_map : UniformContinuous (map f) :=
+  pkg.uniformContinuous_extend
+#align abstract_completion.uniform_continuous_map AbstractCompletion.uniformContinuous_map
+
+theorem continuous_map : Continuous (map f) :=
+  pkg.continuous_extend
```

Comment:
```suggestion
@[continuity]
theorem continuous_map : Continuous (map f) :=
  pkg.continuous_extend
```

Reply:
Does not yet seem to be implemented, but I added a porting note to remind people to add the tag once it is implemented.